src/plugins/discover/public/application/angular/doc_table/components/table_header/table_header_column.tsx

```diff
@@ -108,7 +104,8 @@ export function TableHeaderColumn({
     {
       active: isSortable && typeof onChangeSortOrder === 'function',
       ariaLabel: getSortButtonAriaLabel(),
-      className: btnSortClassName,
+      className: (!sortDirection && 'kbnDocTableHeader__sortChange') || '',
```

Comment:
```suggestion
      className: !sortDirection ? 'kbnDocTableHeader__sortChange' : '',
```

I think this might read a bit more intuitive.